codegen-client/src/test/kotlin/software/amazon/smithy/rust/codegen/client/smithy/endpoint/EndpointsDecoratorTest.kt

```diff
@@ -194,6 +216,8 @@ class EndpointsDecoratorTest {
                                                     .map(ToString::to_string)
                                                     .collect::<Vec<_>>()
                                             )
+                                            .jmes_path_param_string_array(vec!["key2".to_string(), "key1".to_string()])
```

Comment:
**question**: are these `jmes_path_param_*` functions internal or public?

Reply:
Probably `pub` since they're methods on service-specific endpoint params builder.

Reply:
Correct, they are `pub` since they are part of the param builder. Normally they won't have the `jmes_path_param_` prefix, thats just because I named them like that in the model:

```smithy
@operationContextParams(
            JmesPathParamString: {
                path: "nested.field",
            }
            JmesPathParamBoolean: {
                path: "nested.boolField",
            }
            JmesPathParamStringArray: {
                path: "keys(nested.mapField)",
            }
        )
```

We could make them private with the assumption that these should only be set by the generated JMESPath getters, but we allow overriding other values set in the model (mostly those on `staticContextParams`) so it seemed most consistent to keep that behavior.